pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/documentation/ApexDocRule.java

```diff
@@ -113,13 +118,16 @@ public Object visit(ASTMethod node, Object data) {
     @Override
     public Object visit(ASTProperty node, Object data) {
         ApexDocComment comment = getApexDocComment(node);
-        if (comment == null) {
-            if (shouldHaveApexDocs(node)) {
-                addViolationWithMessage(data, node, MISSING_COMMENT_MESSAGE);
-            }
-        } else {
-            if (getProperty(REPORT_MISSING_DESCRIPTION_DESCRIPTOR) && !comment.hasDescription) {
-                addViolationWithMessage(data, node, MISSING_DESCRIPTION_MESSAGE);
+
+        if (getProperty(REPORT_MISSING_PROPERTY_DESCRIPTOR)) {
```

Comment:
I think we should move this check inside "shouldHaveApexDocs" - then reportMissingDescription works like the everywhere else.